src/brogue/SeedCatalog.c

```diff
@@ -1,17 +1,62 @@
+/*
+ *  SeedCatalog.c
+ *  Brogue
+ *
+ *  Copyright 2012. All rights reserved.
+ *
+ *  This file is part of Brogue.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
 #include "Rogue.h"
 #include "IncludeGlobals.h"
 
-static void seedCatalogItemName(item *theItem, char *name, creature *theMonster) {
-    char buf[500] = "", monster[128] = "", location[36] = "", usageLocation[36] = "";
+#define  QUANTITY_1  1
```

Comment:
What's the rationale for this macro?

Reply:
Let me first say that was a nicely worded question. So, ummm... that guy sneaked in without the rest of his family...

```
#define  CSV_HEADER_STRING "dungeon_version,seed,depth,quantity,category,kind,enchantment,runic,vault_number,opens_vault_number,carried_by_monster_name,ally_status_name,mutation_name"
#define  NO_ENCHANTMENT_STRING ""
#define  NO_RUNIC_STRING ""
#define  NO_VAULT_STRING ""
#define  NO_OPENS_VAULT_STRING ""
#define  NO_CARRIED_BY_MONSTER_STRING ""
#define  NO_ALLY_STATUS_STRING ""
#define  NO_MUTATION_STRING ""
```
...which makes it easier to decipher what is going on in the call to printSeedCatalogLine. Here's a snippet from a change I'm working on:

```
        printSeedCatalogCsvLine(rogue.seed, rogue.depthLevel, gold, "gold", kindName, NO_ENCHANTMENT_STRING,
                                NO_RUNIC_STRING, NO_VAULT_STRING, NO_OPENS_VAULT_STRING, NO_CARRIED_BY_MONSTER_STRING,
                                NO_ALLY_STATUS_STRING, NO_MUTATION_STRING);
```
But I can see how QUANTITY_1 is a bit overkill. I'll remove it.

Reply:
I see, that's a nice idea in C's absence of keyword arguments. But yes, I think just using `1` is clear enough

Reply:
Done